emoji-substitution/substitute.js

```diff
@@ -0,0 +1,93 @@
+/*
+ * This file is responsible for performing the logic of replacing
+ * all occurrences of each mapped word with its emoji counterpart.
+ */
+
+// emojiMap.js defines the 'sortedEmojiMap' variable.
+// Referenced here to reduce confusion.
+const emojiMap = sortedEmojiMap;
+
+/*
+ * For efficiency, create a word --> search RegEx Map too.
+ */
+let regexs = new Map();
+for (let word of emojiMap.keys()) {
+  // We want a global, case-insensitive replacement.
+  // @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp
+  regexs.set(word, new RegExp(word, 'gi'));
```

Comment:
Hm, I think especially for short and common patterns like _man_, this sounds a bit too aggressive to me. It's cool when for things like _mankind_, but words that contain _man_ but have no relation to the concept of a _man_ could cause some confusion.